packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -0,0 +1,553 @@
+///   title: "Example Window",
+/// );
+/// runApp(
+///   RegularWindow(
+///     controller: controller,
+///     child: MaterialApp(home: Container()),
+///   ),
+/// );
+/// ```
+///
+/// When a [RegularWindow] widget is removed from the tree, the window that was created
+/// by the [controller] is automatically destroyed if it has not yet been destroyed.
+///
+/// Widgets in the same tree as the [child] widget will have access to the
+/// [RegularWindowController] via the [WindowControllerContext] widget.
+///
+/// {@macro flutter.widgets.windowing.experimental}
+class RegularWindow extends StatefulWidget {
+  /// Creates a regular window widget.
+  /// [controller] the controller for this window
+  /// [child] the content to render into this window
+  /// [key] the key for this widget
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  RegularWindow({super.key, required this.controller, required this.child}) {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_windowingDisabledErrorMessage);
+    }
+  }
+
+  /// Controller for this widget.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  final RegularWindowController controller;
+
+  /// The content rendered into this window.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  final Widget child;
+
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  @override
+  State<RegularWindow> createState() => _RegularWindowState();
+}
+
+class _RegularWindowState extends State<RegularWindow> {
+  @override
+  void dispose() {
+    widget.controller.destroy();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return View(
+      view: widget.controller.rootView,
+      child: WindowControllerContext(controller: widget.controller, child: widget.child),
+    );
+  }
+}
+
+/// Provides descendants with access to the [BaseWindowController] associated with
+/// the window that is being rendered.
+///
+/// {@macro flutter.widgets.windowing.experimental}
+@internal
+class WindowControllerContext extends InheritedWidget {
+  /// Creates a new [WindowControllerContext].
+  ///
+  /// This widget is used by the window widgets to provide
+  /// widgets in a window's subtree with access to information about
+  /// the window.
+  ///
+  /// The [controller] is the controller associated with this window,
+  /// and the [child] is the widget tree that will have access
+  /// to this context.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  WindowControllerContext({super.key, required this.controller, required super.child}) {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_windowingDisabledErrorMessage);
+    }
+  }
+
+  /// The controller associated with this window.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  final BaseWindowController controller;
+
+  /// Returns the [WindowContext] if any
```

Comment:
This method _requires_ that a parent widget has a window controller. We should also add a "maybe" variant in case the caller isn't sure if there's a window controller or not in the parents.

For prior art, see: https://api.flutter.dev/flutter/widgets/MediaQuery/maybeOf.html